Form Field Blocks: Replace dashicon with SVG icons

SirLouen opened the pull request:
Fixes: #55727

This has been stuck for quite a while. It's time to push it forward.

Most of the dashicons have been preserved in SVG format, some others have been changed with current existing SVG icons that perfectly fit. One extra icon has been included (a hash icon for the `Number input`) 

This fix is targeting the `form-input` block, not the `form` block. Another PR will take place just for the `form` block once this has been sorted out.

## Screenshots

|Before|After|
|-|-|
|<img width="345" height="457" alt="image" src="https://github.com/user-attachments/assets/f40830f3-d3e4-4995-b71b-9a7639663909" />|<img width="351" height="414" alt="image" src="https://github.com/user-attachments/assets/e16f411f-e1f8-417b-a804-5a7872f5632d" />
|

SirLouen committed:
Form Field Block: Replace dashicon with SVG icon

t-hamano reviewed (commented):
Thanks for the PR!

As I understand it, this issue cannot move forward yet for two reasons:

- Dashicons cannot be converted to SVG directly. Dash icons are not consistent with Gutenberg's SVG icons, especially at larger sizes. We need to adjust the SVG data.
- There's still no consensus on what the phone icon should be. See #55718

As the issue has the `Needs Design Feedback` label, I think we need the cooperation of the design team to move this issue forward, but I'm not sure they have the bandwidth.

SirLouen commented:
@t-hamano do you know where I can find the guidelines for the ideal measures for SVG? I'm going to ask my designer colleague to fix the required dashicons to perfectly fit in the new SVG format so we can have this moving and don't leave it stale for longer. It has been under feedback pending for 5+ months already, so it's time to move the needle.

I will make an effort to push Forms away from the experimental state.

About the phone thing, I've been reading through it; it's the classic discussion that can be open for years (2 years already). Personally, I like your option of using `mobile` SVG in fact I used this in my first iteration, but since it is currently using the other dashicon (the old phone version), I would rather stick to it, and since it was currently “approved” (as this is how it was released), I would prefer to adapt it to current dimensions, and if #55718 one day gets unstuck, we will send another PR to switch it with whatever decision is taken.

t-hamano commented:
> do you know where I can find the guidelines for the ideal measures for SVG?

There may not be clear guidelines, but I remember there's Figma, which manages all the icons. Maybe you can ask on the `design` channel on Slack.

Another suggestion is to make any new icons internal rather than adding them to `@wordpress/icons`. This is because the Form block is still experimental. Making the icons internal makes it easier to make major changes to the design later, and to add or remove icons.

SirLouen commented:
> All the icons in that Figma already exist in the `@wordpress/icons` package. You can [view them on Storybook](https://wordpress.github.io/gutenberg/?path=/story/icons-icon--library).

Thanks @jameskoster I was looking in the wrong place

https://www.figma.com/design/Qbj4SrtpC6YHncVCnVGmL0/WordPress-icon-exploration?node-id=1-1398&t=nPlg9wq7EQfb2jnU-0

Now I've found the right figma space for the icons. The exact specifications are 24x24 icons with 18x18 max icon size, and regular lines are 1.5px width @t-hamano (FYI). I will make sure the icons proposed fit this and make them internal for now. 

I don't know which is the exact protocol to have them added to that curated list.

jameskoster commented:
Btw could we use the `mobile` icon for the Telehpone Input block?

SirLouen commented:
Ok, let me add the adjustments for the non default icons. Given that this is a single block, probably all 3  could be triaged here. (`envelope`, `forms`, and `hash`), if we are going to be using `mobile` instead

SirLouen commented:
Updated:
- `forms` icon adapted from the original dashicons, with 24×24 canvas and 18×18 icon maintaining the original aspect ratio.
- New `hash` icon standardized with, 24×24 canvas, 18×18 icon, lines 1.5px
- Switched to the `mobile` icon and removed `phone`

Designs ready for icon review 
cc @jameskoster @mattmiklic 

<img width="177" height="228" alt="image" src="https://github.com/user-attachments/assets/3e4b4fe6-501f-477d-ab40-a216436fb8e4" />

jameskoster commented:
I don't know how others feel but my instinct is that we should discuss the new icons in dedicated PRs. The other changes seem good.

SirLouen committed:
Icon: Rename phone icon to mobile and remove SVG file

SirLouen committed:
feat: Add internal icons structure

SirLouen committed:
feat: Refactor form input icons and update variations to use new unified icon structure

SirLouen committed:
fix: Remove unused icon imports and delete obsolete SVG files

SirLouen committed:
fix: Update SVG icon dimensions to use viewBox for better scaling

SirLouen committed:
refactor: update variations to use a single icon reference

t-hamano reviewed (approved):
LGTM!
